libraries/primitive_consul_health.rb

```diff
@@ -30,7 +30,7 @@ def are_checks_passing?(tries)
         checks = Diplomat::Check.checks
 
         non_passing = @options[:checkids]
-          .select { |id| checks[id]['CheckID'] }
+          .map { |id| "service:#{id}" }
```

Comment:
good remark. Indeed @ashangit, some checks can be set on the all agent and not only on the service, could you modify your patch to take that into account?